api/mesh/v1alpha1/dataplane.proto

```diff
@@ -46,6 +46,10 @@ message Dataplane {
     // Public IP on which the dataplane is accessible in the network.
     string address = 5;
 
+    // AdvertiseAddress is the public routable address for the DP in case
+    // address is not routable; envoy binds to address
+    string advertiseAddress = 7;
```

Comment:
Can you explain this more? The "address" field is documented as being publicly accessible, so if this is the real publicly accessible IP, that implies that "address" is set wrong. If there's 2 different publicly accessible addresses, then can there be more than 2? Do certain clients need to pick specific addresses sometime?

Reply:
Hi @jpeach , yes there could be cases where part of the micro-service is running in an overlay network. Address will be allocated from the overlay n/w. Other data-plane will not be able to connect to these data-plane. 
For these cases, we may want to add another advertise address so that other data-plane can connect to this advertise address and there will be additional mapping done to patch advertise address with address. 
ex: docker application running in a private network on a VM.

Reply:
So some DPs will connect to `address` and others will connect to `advertiseAddress`? How does any DP know which one to use?

The explanation of what this field is for and how it changes the behavior of the system needs to go somewhere, preferably i a long block comment above the field.

I expect we can come up with a better name than "advertiseAddress", which sounds like a command rather than a description. Maybe "externalAddress" or "secondaryAddress"?

Reply:
Hi @jpeach dp will always bind to address and advertise address will take precedence in case advertise address is provided. I will add a block comment for this section.

For renaming, should we choose external address because secondary address might confuse user. let me know your opinion I will change it accordingly.

Reply:
@jpeach Let me know if i should rename this variable to externalAddress
@jakubdyszkiewicz

Reply:
> Hi @jpeach dp will always bind to address and advertise address will take precedence in case advertise address is provided. I will add a block comment for this section.

So you only ever bind to one of the two addresses?

Reply:
I like `adverisitedAddress` name because it's consistent with ZoneIngress.
If we were to change this, I think we should do it for both resources.

Reply:
> > Hi @jpeach dp will always bind to address and advertise address will take precedence in case advertise address is provided. I will add a block comment for this section.
> 
> So you only ever bind to one of the two addresses?

Hi @jpeach , DP always bind to address, advertiseAddress is just to update the service entry for this DP in the control plane.

Reply:
> I like `adverisitedAddress` name because it's consistent with ZoneIngress.
> If we were to change this, I think we should do it for both resources.

Hi @jakubdyszkiewicz , yes that is why this name was choose.

Reply:
this should be `advertisedAddress`